src/Microsoft.AspNetCore.SignalR.Client/InvocationSubject.cs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    internal class InvocationSubject : IObservable<object>
+    {
```

Comment:
This looks like deadlock central. Triggering callbacks inside of locks smells like badness to me and will likely end being a large source of errors. Can we instead trigger them outside of the lock after we determine that they need to be called.

Also triggering callbacks inline on subscribe is basically the default tcs behavior which we try to change everywhere. Do we want to dispatch by default. Does user code run directly on these callbacks?